src/test/java/jp/co/yahoo/yosegi/util/TestIndexAndObject.java

```diff
@@ -0,0 +1,172 @@
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+public class TestIndexAndObject {
+
+  @Test
+  public void T_newInstace_1(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 0 );
+  }
+
+  @Test
+  public void T_newInstace_2(){
+    assertThrows( RuntimeException.class ,
+      () -> {
+        IndexAndObject<String> indexAndObj = new IndexAndObject<String>( -1 );
+      }
+    );
+  }
+
+  @Test
+  public void T_getStartIndex_1(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 0 );
+    assertEquals( indexAndObj.getStartIndex() , 0 );
+  }
+
+  @Test
+  public void T_getStartIndex_2(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 100 );
+    assertEquals( indexAndObj.getStartIndex() , 100 );
+  }
+
+  @Test
+  public void T_add_get_1(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 0 );
+    indexAndObj.add( "test0" );
+    assertEquals( indexAndObj.get(0) , "test0" );
+  }
+
+  @Test
+  public void T_add_get_2(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 0 );
+    indexAndObj.add( "test0" );
+    indexAndObj.add( "test1" );
+    assertEquals( indexAndObj.get(0) , "test0" );
+    assertEquals( indexAndObj.get(1) , "test1" );
+  }
+
+  @Test
+  public void T_add_get_3(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 5 );
+    indexAndObj.add( "test0" );
+    assertEquals( indexAndObj.get(5) , "test0" );
+  }
+
+  @Test
+  public void T_add_get_4(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 5 );
+    indexAndObj.add( "test0" );
+    indexAndObj.add( "test1" );
+    assertEquals( indexAndObj.get(5) , "test0" );
+    assertEquals( indexAndObj.get(6) , "test1" );
+  }
+
+  @Test
+  public void T_add_get_5(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 5 );
+    indexAndObj.add( "test0" );
+    assertThrows( ArrayIndexOutOfBoundsException.class ,
+      () -> {
+        indexAndObj.get(4);
+      }
+    );
+  }
+
+  @Test
+  public void T_add_get_6(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 5 );
+    indexAndObj.add( "test0" );
+    assertThrows( IndexOutOfBoundsException.class ,
+      () -> {
+        indexAndObj.get(6);
+      }
+    );
+  }
+
+  @Test
+  public void T_size_1(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 0 );
+    assertEquals( indexAndObj.size() , 0 );
+    indexAndObj.add( "test0" );
+    assertEquals( indexAndObj.size() , 1 );
+    indexAndObj.add( "test1" );
+    assertEquals( indexAndObj.size() , 2 );
+  }
+
+  @Test
+  public void T_size_2(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 5 );
+    assertEquals( indexAndObj.size() , 0 );
+    indexAndObj.add( "test0" );
+    assertEquals( indexAndObj.size() , 1 );
+    indexAndObj.add( "test1" );
+    assertEquals( indexAndObj.size() , 2 );
+  }
+
```

Comment:
Please describe what are tested in the test method separately at T_size_get_?.
For example, T_size_xxx or some.

Reply:
Please add a prefix or suffix like T_ or Text or some so that the code reader knows it is a test method including assertThat or assertThrows and/or some testing methods.